### PR TITLE
refactor: SAS Token Generation

### PR DESCRIPTION
**Pull request by tombuildsstuff**
This PR switches over to using SAS Token Generation from the package in the split-out library - introduced in https://github.com/hashicorp/go-azure-helpers/pull/8

~I'll update the vendoring once that PR is merged~ done

### CODE REVIEW

File: azurerm/data_source_storage_account_sas.go [hunk 1]
```diff
@@ -199,7 +194,7 @@ func dataSourceArmStorageAccountSasRead(d *schema.ResourceData, _ interface{}) e
 	permissions := buildPermissionsString(permissionsIface[0].(map[string]interface{}))
 
 	// Parse the connection string
-	kvp, err := parseAzureStorageAccountConnectionString(connString)
+	kvp, err := storage.ParseStorageAccountConnectionString(connString)
```
[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply azurerm/data_source_storage_account_sas.go]: There’s two types of SAS token - an account token and a service token; this is mostly to signify that - so perhaps we want to rename the Compute method?

**Comment by tombuildsstuff** [comment]
Tests pass:

screenshot 2018-11-26 at 16 21 43

File: azurerm/data_source_storage_account_sas.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (katbyte) [reply azurerm/data_source_storage_account_sas.go]: Ah that makes sense, i'd 100% vote to change the CompureMethold

[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply azurerm/data_source_storage_account_sas.go]: will handle this in the other lib

**Pull request merged**